viabel/optimization.py

```diff
@@ -151,6 +166,11 @@ def __init__(self, learning_rate, *, weight_decay=0, beta=0.9, jitter=1e-8,
         super().__init__(learning_rate, weight_decay=weight_decay, diagnostics=diagnostics)
 
     def reset_state(self):
+        """
+        resetting the state
```

Comment:
Please describe in more detail what is actually done – that is, what does "state" actually mean here?